server.go

```diff
@@ -462,7 +461,15 @@ func (p *sshFxpOpenPacket) respond(svr *Server) responsePacket {
 		osFlags |= os.O_EXCL
 	}
 
-	f, err := os.OpenFile(svr.toLocalPath(p.Path), osFlags, 0o644)
+	mode := os.FileMode(0o644)
+	// Like OpenSSH, we only handle permissions here, if the file is being created.
+	// Otherwise, the permissions are ignored.
```

Comment:
The semantic line break here is actually between the first line before the `if`. It should be a full stop not a comma.